packet.go

```diff
@@ -0,0 +1,217 @@
+// Checksum computations based on logic in github.com/google/gopacket
+//
+// Copyright 2012 Google, Inc. All rights reserved.
+// Copyright 2009-2011 Andreas Krennmair. All rights reserved.
+
+package gonat
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/getlantern/errors"
+)
+
+// TCPFlags are the different flags supported in the TCP header
+const (
+	TCPFlagRST = 0x04
```

Comment:
Why export this?

Reply:
This IP packet parsing is potentially useful outside of gonat. Not using it right now, but it seems like a public-ish API.

Reply:
I see.  Thanks!